### PR TITLE
Add CL2 test jobs for KOPS scale test

### PR DESCRIPTION
**Pull request by prateekgogia**
Part of this effort -https://github.com/kubernetes/test-infra/issues/29139
Also corresponding PR for KOPS for running this test - https://github.com/kubernetes/kops/pull/15538

### CODE REVIEW

File: config/jobs/kubernetes/kops/templates/presubmit-scenario.yaml.jinja [hunk 1]
```diff
@@ -20,6 +20,11 @@
     decoration_config:
       timeout: {{job_timeout}}
     path_alias: k8s.io/kops
+    extra_refs:
+    - org: kubernetes
+      repo: perf-tests
+      base_ref: master
+      path_alias: k8s.io/perf-tests
```
Comment (hakman) [inline config/jobs/kubernetes/kops/templates/presubmit-scenario.yaml.jinja:27]: This is specific to perf-tests, but the template is for all kOps pre-submits. Maybe check if scenario is "scalability"?

File: config/jobs/kubernetes/kops/build_jobs.py [hunk 2]
```diff
@@ -1060,6 +1062,16 @@ def generate_presubmits_scale():
...
+        ),
+        presubmit_test(
+            name='presubmit-kops-aws-scale-amazonvpc-using-cl2',
+            scenario='scalability',
+            networking='amazonvpc',
+            always_run=False,
+            artifacts='$(ARTIFACTS)',
+            env={
+                'RUN_CL2_TEST': "true",
+            }
```
Comment (hakman) [inline config/jobs/kubernetes/kops/build_jobs.py:1074]: `artifacts` ends up being an env var in the end, same as `RUN_CL2_TEST`. Why add an extra parameter for it?

**Review by hakman — CHANGES_REQUESTED** [review]

**Comment by dims** [comment]
/approve

( will leave /lgtm to @hakman )

File: config/jobs/kubernetes/kops/build_jobs.py (hunk 2 above)
Comment (prateekgogia) [reply config/jobs/kubernetes/kops/build_jobs.py:1074]: Initially, I tried adding it as an env variable but the result in the job generated has 
```
- name: ARTIFACTS
   value: "$(ARTIFACTS)"
```
I compared it with other job definitions and to keep it consistent `value: $(ARTIFACTS)` went with this approach of defining a placeholder in the template and hence an extra param.

File: config/jobs/kubernetes/kops/templates/presubmit-scenario.yaml.jinja (hunk 1 above)
Comment (prateekgogia) [reply config/jobs/kubernetes/kops/templates/presubmit-scenario.yaml.jinja:27]: Thanks good call, I have added this check.

**Review by hakman — APPROVED** [review]

**Pull request merged**